### PR TITLE
ED-5959 Correct image scaling

### PR DESCRIPTION
**Pull request by sheikrxnt**

### CODE REVIEW

**Review by kalimuthu-rxnt — APPROVED** [review]

File: src/MultipleComplaintImageFactory.js [hunk 1]
```diff
@@ -75,7 +75,11 @@ export default function(Complaints, SelectImages) {
           let { active } = state;
           let activeObject = state[active];
           let complaints = activeObject.complaints.concat({ pos: pointerPos });
-          let updatedItem = { [active]: { ...activeObject, complaints } };
+          let currentWidth = this.currentWidth;
+          let updatedItem = {
+            [active]: { ...activeObject, complaints },
+            srcWidth: currentWidth
```
Comment (sneu012) [inline src/MultipleComplaintImageFactory.js:81]: Do we only need to store the width?

File: src/MultipleComplaintImageFactory.js [hunk 2]
```diff
@@ -203,6 +222,26 @@ export default function(Complaints, SelectImages) {
...
+      if (
+        srcWidth !== "" &&
+        this.currentWidth !== srcWidth &&
+        complaints.length > 0
+      ) {
+        complaints.forEach(item => {
+          let positionX = "";
+          let positionY = "";
+          positionX = item.pos.x * (this.currentWidth / srcWidth);
+          positionY = item.pos.y * (this.currentWidth / srcWidth);
```
Comment (sneu012) [inline src/MultipleComplaintImageFactory.js:240]: This logic assumes the image is always gonna have the same width and height. Is the true? Have you tried using multiple images with different dimensions?

File: src/MultipleComplaintImageFactory.js (hunk 1 above)
Comment (sheikrxnt) [reply src/MultipleComplaintImageFactory.js:81]: Yes. we are using width and height both as width only.

```
stage.width(width);
stage.height(width);
```

```
image.width(width - imagePadding);
image.height(width - imagePadding);

```

File: src/MultipleComplaintImageFactory.js (hunk 2 above)
Comment (sheikrxnt) [reply src/MultipleComplaintImageFactory.js:240]: The width changes according to the resolutions. Tried with different images with different dimensions all looks good.

Based on the current width and previously stored width the x and y positions are placed.

File: src/MultipleComplaintImageFactory.js [hunk 3]
```diff
@@ -203,6 +222,26 @@ export default function(Complaints, SelectImages) {
       let image = this.refs.image;
       image.width(width - imagePadding);
       image.height(width - imagePadding);
+
+      let { active } = this.state;
+      let complaints = this.state[active] ? this.state[active].complaints : [];
+      let srcWidth =
+        this.state.srcWidth !== undefined ? this.state.srcWidth : "";
```
Comment (sneu012) [inline src/MultipleComplaintImageFactory.js:229]: The number is being defaulted to a string.

**Review by sneu012 — APPROVED** [review]
Looks good.

File: src/MultipleComplaintImageFactory.js (hunk 3 above)
Comment (sheikrxnt) [reply src/MultipleComplaintImageFactory.js]: Done

File: src/MultipleComplaintImageFactory.js [hunk 4]
```diff
@@ -405,14 +449,24 @@ class Drawing extends Component {
     canvas.height = this.props.parentImageSize;
     const context = canvas.getContext("2d");
     let allPointsArray = [];
-
+    const srcWidth =
+      this.props.srcWidth !== undefined ? this.props.srcWidth : "";
```
[the remark this answers is not in the document]
Comment (sheikrxnt) [reply src/MultipleComplaintImageFactory.js]: done

**Review by sneu012 — APPROVED** [review]

**Pull request merged**